implementations/rust/ockam/ockam/src/credential/exchange.rs

```diff
@@ -150,12 +289,43 @@ impl<S: AuthenticatedStorage, V: IdentityVault> CredentialExchangeWorker<S, V> {
         let path_segments = req.path_segments::<5>();
         let method = match req.method() {
             Some(m) => m,
-            None => todo!(),
+            None => {
+                return Ok(Response::bad_request(req.id())
+                    .body("Invalid method")
+                    .to_vec()?)
+            }
         };
 
         let r = match (method, path_segments.as_slice()) {
             (Post, ["actions", "present"]) => {
-                self.receive_presented_credential(req, sender, dec).await?
+                let credential: Credential = dec.decode()?;
+
+                CredentialExchange::receive_presented_credential(
+                    req.id(),
+                    req.path(),
+                    sender,
+                    credential,
+                    &self.authorities,
+                    None,
+                    &self.vault,
+                    &self.authenticated_storage,
+                )
+                .await?
+            }
+            (Post, ["actions", "present_mutual"]) => {
+                let credential: Credential = dec.decode()?;
+
+                CredentialExchange::receive_presented_credential(
+                    req.id(),
+                    req.path(),
+                    sender,
+                    credential,
+                    &self.authorities,
+                    self.present_back.as_ref(),
```

Comment:
If this is `None` it will just send an OK response without the credential body. Should we not error in this case?

Reply:
Interesting question. So, if initiator presents credential for us and expects credential in return, but we don't have it, should we accept his credential or not? I think it should accept it. Initiator will error either way, though error may be not super-friendly atm, there is space for improvement